### PR TITLE
feat(export): Add parentship for hosts

### PR DESCRIPTION
**Pull request by Guillaume28**
Add host parents in the export.
Related to issue #5605

### CODE REVIEW

File: www/class/centreon-clapi/centreonHost.class.php [hunk 1]
```diff
@@ -885,6 +885,60 @@ public function __call($name, $arg)
...
+            $parentShip[$secondKey][] = $firstkey;
+        }
+
+        $hosts = array();
+        foreach ($elements as $element){
+            $hosts[] = $element['host_id'];
+        }
+
+        $sortedHosts = array();
+        while($hostId = array_pop($hosts)){
```
Comment (kduret) [inline www/class/centreon-clapi/centreonHost.class.php]: PSR-2 please :)

File: www/class/centreon-clapi/centreonHost.class.php [hunk 2]
```diff
@@ -885,6 +885,60 @@ public function __call($name, $arg)
...
+     * @return array
+     */
+    protected function getHostListByParent(&$elements)
+    {
+        $hostParent = new \Centreon_Object_Relation_Host_Parent_Host();
+
+        $parentShip = array();
+        $relations = $hostParent->getRelations();
+        foreach ($relations as $relation) {
+            $firstkey = $relation[$hostParent->getFirstKey()];
```
Comment (loiclau) [inline www/class/centreon-clapi/centreonHost.class.php]: camelcase

File: www/class/centreon-clapi/centreonHost.class.php [hunk 3]
```diff
@@ -885,6 +885,60 @@ public function __call($name, $arg)
...
+        }
+
+        $hosts = array();
+        foreach ($elements as $element) {
+            $hosts[] = $element['host_id'];
+        }
+
+        $sortedHosts = array();
+        while ($hostId = array_pop($hosts)) {
+            if(!in_array($hostId, array_keys($parentShip))){
```
Comment (loiclau) [inline www/class/centreon-clapi/centreonHost.class.php]: space

File: www/class/centreon-clapi/centreonHost.class.php [hunk 4]
```diff
@@ -885,6 +885,60 @@ public function __call($name, $arg)
...
+                $sortedHosts[] = $hostId;
+            } else {
+                $parents = $parentShip[$hostId];
+                $parentExported = true;
+                foreach($parents as $parentId){
```
Comment (loiclau) [inline www/class/centreon-clapi/centreonHost.class.php]: space

File: www/class/centreon-clapi/centreonHost.class.php [hunk 5]
```diff
@@ -885,6 +885,60 @@ public function __call($name, $arg)
...
+                    if(!in_array($parentId, $sortedHosts)){
```
Comment (loiclau) [inline www/class/centreon-clapi/centreonHost.class.php]: space

**Comment by loiclau** [comment]
can't work on master

**Comment by Guillaume28** [comment]
Another pull request was created on the master branch for the whole thing to work.

**Pull request merged**